pkg/kyverno/test/test_command.go

```diff
@@ -573,32 +595,181 @@ func getFullPath(paths []string, policyResourcePath string, isGit bool) []string
 }
 
 func applyPoliciesFromPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
+	versionedvalues := &versionedTest{}
+
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
+		return sanitizederror.NewWithError("failed to decode yaml", err)
+	}
+
+	// check for old format
+	if versionedvalues.APIVersion == "" {
+		err = nonVersionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	} else {
+		err = versionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	}
+
+	return err
+
+}
+
+func versionedPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
 	openAPIController, err := openapi.NewOpenAPIController()
+	if err != nil {
+		return sanitizederror.NewWithError("failed to get new openAPIcontroller", err)
+	}
 	engineResponses := make([]*response.EngineResponse, 0)
 	var dClient *client.Client
-	values := &Test{}
+	versionedvalues := &versionedTest{}
```

Comment:
I don't quite understand why we declare all variables here instead of when they are needed. But this was the case in the old implementation as well.

Reply:
I was under the impression that in general, the format is to declare all variables of function together at the beginning.

Reply:
This is not done in golang usually. 
Variables should be declared when they are needed with the minimal scope required. E.g. if a variable is needed in a for loop, variable declaration should happen just before the for loop.